docs/DeveloperGuide.md

```diff
@@ -743,38 +834,96 @@ be able to view the applicant from the list.
+* 1a. Input format is wrong.
 
-    * Step 4 repeats as many times as there are requests.
+    * 1a1. TrackAScholar displays an error message prompting correct input format. <br>
+      Use case resumes at step 1.
 
-* 1b. TrackAScholar detects an error in the entered applicant status/detects no applicant status input.
+* 1b. Input parameters is invalid.
 
-    * 1b1. TrackAScholar requests for the correct applicant status.
-    * 1b2. User enters new applicant status.
-    * Steps 1b1-1b2 are repeated until the applicant status entered matches a valid existing applicant status.
-    * Use case resumes from step 2.
+    * 1b1. TrackAScholar displays an error message showing parameter requirements. <br>
+      Use case resumes at step 1.
+
+--------------------------------------------------------------------------------------------------------------------
 
+**Use case: UC10 - Sorts all applicants by name, scholarship or application status**
 
-*{More to be added}*
+**MSS**
+
+1. User requests a sorted list sorted by name, scholarship or application status.
+2. TrackAScholar shows the sorted list of applicants with the specified input parameter. <br>
+   Use case ends.
+
+**Extensions**
+
+* 1a. Input format is wrong.
+
+    * 1a1. TrackAScholar displays an error message prompting correct input format. <br>
+      Use case resumes at step 1.
+
+* 1b. Input parameters is invalid.
+
+    * 1b1. TrackAScholar displays an error message showing parameter requirements. <br>
+      Use case resumes at step 1.
+
+--------------------------------------------------------------------------------------------------------------------
+
+**Use case: UC11 - Imports a json file into TrackAScholar**
+
+**Guarantees:**
+* After merging, no two applicants will have the same name.
+
+**MSS**
+
+1. User imports a json file into TrackAScholar and specifies the type of merging.
+2. TrackAScholar merges the imported json file with the current file.
+3. TrackAScholar shows the lists of all applicants after importing.
+   Use case ends.
+
+**Extensions**
+  
+* 1a. Input format is wrong.
+
+    * 1a1. TrackAScholar displays an error message prompting correct input format. <br>
+      Use case resumes at step 1.
+
+* 1b. Input parameters is invalid.
+
+    * 1b1. TrackAScholar displays an error message showing parameter requirements. <br>
+      Use case resumes at step 1.
+
+* 1c. Imported file does not exist.
+
+    * 1c1. TrackAScholar displays an error message informing the user that there is no file to be imported. <br>
+      Use case resumes at step 1.
+
+* 1d. Imported file is not a json file.
+
+    * 1d1. TrackAScholar displays an error message showing the imported file type requirements. <br>
+      Use case resumes at step 1.
+
+* 2a. An imported applicant has the same name as an applicant already in the applicant list.
+
+    * 2a1. TrackAScholar merges the applicants according to the type of merging specified by the user input. <br>
```

Comment:
Not sure whether need to specify the type of merging here?

Reply:
not specified as use cases should not go into detail implementation wise